### PR TITLE
Presentation aks fix: The name displayed for the storage is inreality everything before -hcp-, this fix helps clearing that confusion.

### PR DESCRIPTION
**Pull request by Tatsinnit**
Detail this PR trying to achieve: 

3 changes:.

* Fix `Pylint` failures.

*  Right representation of the Storage URL matching the blobname which gets created. More detail could be found here: https://github.com/Azure/aks-periscope/issues/12 and currently when the storage blob gets created its following the convention under following ARM convention : https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/error-storage-account-name Hence when using persicope user navigate to `storage explorer` they will see blobname you will see logs uploaded under the blob container: **example only** `aksaspen2-dns-22701a84` and not `aksaspen2-dns-22701a84-hcp-eastus2-azmk8s-io`

* The `return` is not necessary to be explicit in python, AFAIK. essentially `return` or `return None` or jsut not mentioning is same. (Unless there is something specific reason, hence detailing here)

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by Tatsinnit** [comment]
Tag: `AKS` 

Ping reviewer /assign : @xizhamsft + @JunSun17 , Thanks heaps!

**Comment by yonzhan** [comment]
S167

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -1500,7 +1500,21 @@ def aks_kollect(cmd,    # pylint: disable=too-many-statements,too-many-locals
     else:
         display_diagnostics_report(temp_kubeconfig_path)
 
-    return
+
+def _trim_fqdn_name_containing_hcp(normalized_fqdn: str) -> str:
```
Comment (gtracer) [inline src/aks-preview/azext_aks_preview/custom.py]: Can we please move to this to a helper file

File: src/aks-preview/azext_aks_preview/custom.py [hunk 2]
```diff
@@ -1500,7 +1500,21 @@ def aks_kollect(cmd,    # pylint: disable=too-many-statements,too-many-locals
...
+    """
+    Trims the storage blob name and takes everything prior to "-hcp-".
+    Currently it is displayed wrong: i.e. at time of creation cli has
+    following limitation:
+    https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/
```
Comment (gtracer) [inline src/aks-preview/azext_aks_preview/custom.py]: To be specific, you are saying that the problems are:
1. the `-` in the name are causing it to fail 
2. having hcp in every name is not ideal as it leads to less characters that are truly unique in the name 

Is that right?

**Comment by JunSun17** [comment]
LGTM, just make sure you have tested it.

File: src/aks-preview/azext_aks_preview/custom.py (hunk 2 above)
Comment (xizhamsft) [reply src/aks-preview/azext_aks_preview/custom.py]: @gtracer , the issue here is that the aks-periscope pods will create a blob for MC with name 'xxx-xxx-hcp-xxxx' like this "xxx-xxx". Basically removing everything after "-hcp" (including -hcp). But our CLI code still uses the full MC name in the URL. Hence the mismatch.

**Comment by yungezz** [comment]
hi @Tatsinnit could you pls add tests to it? thanks.

File: src/aks-preview/azext_aks_preview/custom.py (hunk 1 above)
Comment (Tatsinnit) [reply src/aks-preview/azext_aks_preview/custom.py]: @gtracer Cool, sounds good, i have moved this to helper:

* Moving fwd plan is to refactor this file. Essentially, I have a plan to do another PR with some refactor for `custom.py` move rest of the helper `functs` out.

**Comment by Tatsinnit** [comment]
:bulb: Test et. al. added, like mentioned I would do another drop next for the generic things I see can be corrected for `aks-preview`. Thanks + ping: cc: @yungezz + @gtracer 

* Also `pylint` was broken which is fixed 

Below was the lint failure.

FYI: 

image

**Pull request merged**